test.js

```diff
@@ -49,27 +49,42 @@ test('is mobile', function (t) {
 })
 
 describe('ua-bruteforce', function () {
-  const limit = 300
-  const checks = {
-    mobile: { result: true },
-    tablet: { result: true, tablet: true },
-    desktop: { result: false }
+  const ua = deviceCategory => {
+    if (!ua[deviceCategory]) {
+      // The same user-agent string belongs to both `desktop` and `mobile` type entries. No chance to detect `deviceType` properly.
+      // https://github.com/intoli/user-agents/blob/867e318bc00880ae00437e5e8efaa8e5e7ac0696/src/user-agents.json.gz
+      // user-agents v1.0.843
+      const exclude =
+        'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.198 Safari/537.36'
+      ua[deviceCategory] = new UserAgent([
```

Comment:
this is still caching the user agent instance 🤔

Reply:
It caches the data preset which is attached to the instance. There's no other way to bind filtered values: `setCumulativeWeightIndexPairs` declared as module scope variable.
https://github.com/intoli/user-agents/blob/master/src/user-agent.js#L140
```js
 random = () => {
    const userAgent = new UserAgent();
    setCumulativeWeightIndexPairs(userAgent, this.cumulativeWeightIndexPairs);
    userAgent.randomize();
    return userAgent;
  };
```

Authors literally suggest using this approach:
https://github.com/intoli/user-agents#useragentrandom
```js
// Use the `random()` method to construct a second user agent.
const firstUserAgent = new UserAgent(filters);
const secondUserAgent = firstUserAgent.random();
```

![image](https://user-images.githubusercontent.com/5288046/143015724-70213f1b-70d6-440b-b1ab-af01d61c9d75.png)

Reply:
I'm not against using .random(), I'm against caching UserAgent instances in ua[...]

Reply:
I've just noticed `new Proxy`. optimizations don't make sense)